### PR TITLE
Aem p4

### PR DESCRIPTION
**Pull request by krystian-hebel**

### CODE REVIEW

**Review by SergiiDmytruk — COMMENTED** [review]
Left some comments primarily on wording.

**Comment by miczyg1** [comment]
I have nothing more to add besides what Sergii pointed out already.

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
+---
+title: TrenchBoot Anti Evil Maid - Phase 4
+abstract: "This blog post marks completion of next phase of TrenchBoot Anti Evil
+           Maid project for Qubes OS. This time the project focused on AMD
+           platforms, which is something that wasn't possible with original
+           solution based on tboot."
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md]: https://github.com/3mdeb/news-and-ideas/pull/592/commits/0ba19895588a5b13c46aad9b806ddba1a08e19b5

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+good reading to catch up on what the TrenchBoot AEM project is about.
+
+## What's new in TrenchBoot AEM?
+
+The most significant change is obviously the ability to use AEM on AMD
+platforms. Contrary to TXT, which uses SINIT ACMs signed by Intel, on AMD the
+first piece of code executed after Dynamic Launch Event is fully controlled by
+the user. Other than two 16-bit fields at the very beginning, format of DCE
+(DRTM Configuration Environment) is decided by the implementation. Most notably,
+this includes hand-offs (both GRUB2->DCE and DCE->Xen) and set and order of
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md]: https://github.com/3mdeb/news-and-ideas/pull/592/commits/0ba19895588a5b13c46aad9b806ddba1a08e19b5

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+decided to not do this.
+
+The DCE implementation used by this project is called Secure Kernel Loader (or
+SKL), and the source code can be found [here](https://github.com/TrenchBoot/secure-kernel-loader).
+
+Unfortunately, TCG, Intel and AMD all use different names for the same component
+or event. Hopefully, the following table can help with navigating between those
+names and abbreviations.
+
+| TCG                                              | Intel                                 | AMD                  | What is it in AEM world                                    |
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md]: https://github.com/3mdeb/news-and-ideas/pull/592/commits/0ba19895588a5b13c46aad9b806ddba1a08e19b5

TrenchBoot tries to stay agnostic, so it had to choose TCG (or introduce it's own terms).

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+Now it should be possible to download and install packages from our repository.
+As some of the packages are also available in standard QubesOS repositories,
+potentially in never versions, those must be temporarily disabled during
+invocation of `qubes-dom0-update`, as shown in the following commands.
+
+> If any of the packages that are part of AEM are updated in standard repos,
+> you will have to choose between using new versions or having working AEM, at
+> least until new AEM release is published or the code gets merged upstream.
+> If you decide to keep AEM after an update broke it, you will have to repeat
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md]: https://github.com/3mdeb/news-and-ideas/pull/592/commits/0ba19895588a5b13c46aad9b806ddba1a08e19b5

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+> reinstalling, you have to update mentioned packages by changing
+> `--action=reinstall` to `--action=update` in the call. If there are multiple
+> packages mentioned, all of them must be updated in the same invocation of
+> `qubes-dom0-update`.
+>
+> Both of those problems will disappear in time - firstly when current `testing`
+> lands in main repo, and then permanently after AEM based on TrenchBoot is
+> fully merged to QubesOS. Until that happens, bear with us.
+
+When booting on legacy (non-UEFI) system, GRUB2 must be manually installed to
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md]: https://github.com/3mdeb/news-and-ideas/pull/592/commits/0ba19895588a5b13c46aad9b806ddba1a08e19b5

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+the MBR.
+
+```bash
+sudo grub2-install /dev/sda
```
[the remark this answers is not in the document]
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md:201]: https://github.com/3mdeb/news-and-ideas/pull/592/commits/0ba19895588a5b13c46aad9b806ddba1a08e19b5

Also added information about installing on disk and not on partition.

**Review by SergiiDmytruk — APPROVED** [review]

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+To install, you have to first add a new repository and import a public part of
+a key that was used to sign RPM packages. To add a new repository, create in
+dom0 as root `/etc/yum.repos.d/aem.repo` with the following content:
+
+```text
+[aem]
+name = Anti Evil Maid based on TrenchBoot
+baseurl = https://dl.3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_v0.4
+gpgcheck = 1
+gpgkey = https://dl.3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_v0.4/RPM-GPG-KEY-aem
```
Comment (krystian-hebel) [inline blog/content/post/2024-04-11-aem_phase4.md:108]: URLs will change!

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+title: TrenchBoot Anti Evil Maid - Phase 4
+abstract: "This blog post marks the completion of the next phase of TrenchBoot
+           Anti Evil Maid project for Qubes OS. This time the project focused
+           on AMD platforms, which is something that wasn't possible with the
+           original solution based on tboot."
+cover: /covers/trenchboot-logo.png
+author: krystian.hebel
+layout: post
+published: true
+date: 2024-04-11
```
Comment (BeataZdunczyk) [inline blog/content/post/2024-04-11-aem_phase4.md:11]:
```suggestion
date: 2024-05-17
```

File: blog/content/post/2024-04-11-aem_phase4.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+```bash
+qvm-run --pass-io sys-net 'curl -L https://dl.3mdeb.com/rpm/QubesOS/r4.2/current/dom0/fc37/RPM-GPG-KEY-tb-aem' > RPM-GPG-KEY-tb-aem
+sudo rpm --import RPM-GPG-KEY-tb-aem
+```
+
+![Importing signing key](/img/qubes_aem_import_key.png)
+
+Now it should be possible to download and install packages from our repository.
+As some of the packages are also available in standard QubesOS repositories,
+potentially in never versions, those must be temporarily disabled during
```
Comment (BeataZdunczyk) [inline blog/content/post/2024-04-11-aem_phase4.md:125]:
```suggestion
potentially in newer versions, those must be temporarily disabled during
```

**Review by BeataZdunczyk — CHANGES_REQUESTED** [review]

File: blog/content/post/2024-04-11-aem_phase4.md (hunk 9 above)
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md:125]: :man_facepalming:

File: blog/content/post/2024-04-11-aem_phase4.md (hunk 7 above)
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md]: Updated in https://github.com/3mdeb/news-and-ideas/pull/592/commits/a6ff9041f440175591c72e7cc7c9f4ab7706411e and https://github.com/3mdeb/news-and-ideas/pull/592/commits/f7b994439be0183065e0a322e8aa202dcf2219b9

File: blog/content/post/2024-04-11-aem_phase4.md (hunk 9 above)
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md]: https://github.com/3mdeb/news-and-ideas/pull/592/commits/36bcaf6c3b67070087da7359d81493343e171ad8

File: blog/content/post/2024-04-11-aem_phase4.md (hunk 8 above)
Comment (krystian-hebel) [reply blog/content/post/2024-04-11-aem_phase4.md]: https://github.com/3mdeb/news-and-ideas/pull/592/commits/36bcaf6c3b67070087da7359d81493343e171ad8

**Review by BeataZdunczyk — APPROVED** [review]

**Pull request merged**